TodoList/Tests/TodoList.Api.Tests/Controllers/ListItemsControllerTest.cs

```diff
@@ -1,192 +1,232 @@
+            
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(actualListItems, Is.EqualTo(expectedListItems).UsingListItemComparer());
+        }
+
+        [TestCase("")]
+        [TestCase("    ")]
+        [TestCase(null)]
+        public void PostAsync_ItemWithNoText_ReturnsCorrectErrorResponse(string postedText)
+        {
+            var expectedKeys = new[] { "Text" };
+            var postedListItem = new ListItem { Id = Guid.Empty, Text = postedText };
+
+            var actionResult = _controller.PostAsync(postedListItem).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.IsNull(actualLocation);
+        }
+
+        [Test]
+        public void PostAsync_ItemWithNonEmptyGuid_ReturnsCorrectErrorResponse()
+        {
+            var expectedKeys = new[] { "Id" };
+            var postedListItem = new ListItem { Id = _guidOfFirstItem, Text = "text" };
+
+            var actionResult = _controller.PostAsync(postedListItem).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.IsNull(actualLocation);
+        }
+
+        [Test]
+        public void PostAsync_WithNullArguments_ReturnsCorrectErrorResponse()
+        {
+            var expectedKeys = new[] { string.Empty };
+
+            var actionResult = _controller.PostAsync(null).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.That(actualLocation, Is.EqualTo(null));
+        }
+
+        [Test]
+        public void PostAsync_WithValidArguments_ReturnsCorrectResponse()
+        {
+            var expectedListItem = new ListItem { Id = _guidOfFirstItem, Text = "text" };
+            var postedListItem = new ListItem { Id = Guid.Empty, Text = "newText" };
+            var expectedLocation = new Uri($"api/v1/ListItems/{_guidOfFirstItem}", UriKind.Relative);
```

Comment:
It would be great to use a different path in tests to prevent "hard-coding" errors (e.g. `some/test/path/{_guidOfFirstItem}/with/sufix` or something)